mart/utils/modality_dispatch.py

```diff
@@ -0,0 +1,58 @@
+#
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+from __future__ import annotations
+
+from typing import Any, Callable
+
+import torch
+from torch import Tensor
+
+__all__ = ["modality_dispatch"]
+
+
+def modality_dispatch(
```

Comment:
If you want to get really fancy you could use [functools.singledispatch](https://docs.python.org/3.9/library/functools.html#functools.singledispatch).

Reply:
Does this obviate the need for `Projector` and `Composer` to be batch aware? If so, I would remove that code.

Reply:
Can you remove the batch code from `Projector` and `Composer` then?